спринт 10 на проверку

PavelRomashin opened the pull request.

PavelRomashin committed:
спринт 10 на проверку

SeveNChaK commented on app/src/main/java/com/example/playlistmaker/TrackHolder.kt:
```diff
@@ -0,0 +1,27 @@
+            .load(model.artworkUrl100)
+            .placeholder(R.drawable.close)
+            .centerCrop()
+            .transform(RoundedCorners(20))
```
⚠️ Здесь радиус задается в пикселях. Поэтому значение из макета (которое мы берем как значение в `dp`) необходимо преобразовать в пиксели.
Удобнее всего будет вынести это значение в ресурсы, а потом достать его здесь: `itemView.resources.getDimensionPixelSize(нужный ресурс)`

И обрати внимание, что в макете он равен 2.

SeveNChaK commented on app/src/main/java/com/example/playlistmaker/TrackHolder.kt:
```diff
@@ -0,0 +1,27 @@
+        trackLength.text = model.trackTime
+        Glide.with(trackImage)
+            .load(model.artworkUrl100)
+            .placeholder(R.drawable.close)
```
⚠️ Плейсхолдер в макеет другой
<img width="340" alt="image" src="https://user-images.githubusercontent.com/22057568/219444871-367bbed9-1073-41fd-a90c-7083c31808d7.png">

SeveNChaK commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -55,5 +55,12 @@
+            android:id="@+id/recyclerView"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"/>
```
Если в разметке задаешь лейаут менеджер, то в коде не надо

SeveNChaK commented on app/src/main/res/layout/track_card.xml:
```diff
@@ -0,0 +1,45 @@
+
+            <TextView
+                style="@style/track_details_style"
+                android:id="@+id/ArtistName"/>
```
⚠️ Если будет длинное название артиста (оно точно будет, когда в следующих спринтах будете грузить данные из интернета), то оно вытолкнет время и верстка сильно разъедется
<img width="410" alt="image" src="https://user-images.githubusercontent.com/22057568/219446194-d00b508c-384a-4f42-b345-1ab433eb52c6.png">

SeveNChaK commented on app/src/main/res/layout/track_card.xml:
```diff
@@ -0,0 +1,45 @@
+
+        <TextView
+            style="@style/track_name_style"
+            android:id="@+id/TrackName"/>
```
⚠️ Отступ между название трека и исполнителем не соответствует макету
Сейчас
<img width="227" alt="image" src="https://user-images.githubusercontent.com/22057568/219446564-f81e04cb-7095-41e3-94d3-78044922449d.png">

В макете
<img width="283" alt="image" src="https://user-images.githubusercontent.com/22057568/219446606-e7fc1a24-7866-4ccc-a995-eda1c55110cc.png">

⚠️ Также тексты должны быть выровнены по середине фотографии. Сейчас название сильно уезжает вниз

SeveNChaK commented on app/src/main/res/layout/track_card.xml:
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout
```
⚠️ Отступ между треками не соответствует макету

SeveNChaK commented on app/src/main/res/layout-v28/track_card.xml:
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout
```
⏫ Можешь попробовать использовать `ConstraintLayout` с ним долно быть проще реализовать картчоку трека и сделать все нужные выравнивания

PavelRomashin committed:
отработка замечаний к дз10:
1) Установлена правильная заглушка(изображение) при отсутствии интернета.
2) переделано скругление иконки трека.
3) убран вызов менеджера слоёв в классе SearchActivity.kt
4) исправлена верстка

SeveNChaK commented on app/src/main/res/layout/track_card.xml:
```diff
@@ -1,38 +1,60 @@
+                android:id="@+id/ArtistName"
                 style="@style/track_details_style"
-                android:id="@+id/ArtistName"/>
+                android:layout_width="fill_parent"
```
`fill_parent` устаревшее значение, вместо него используется `match_parent`. Но тут и оно не нужно, тут лучше `wrap_content`

SeveNChaK commented on app/src/main/res/layout/track_card.xml:
```diff
@@ -41,5 +63,6 @@
     <ImageView
```
⚠️ Отступ между стрелочкой и данными о треке в макете другой
Сейчас значительно больше
<img width="469" alt="image" src="https://user-images.githubusercontent.com/22057568/219954594-3ba7179a-0fdf-4ced-97b9-3d9b85426977.png">

Макет
<img width="537" alt="image" src="https://user-images.githubusercontent.com/22057568/219954576-d93104d6-7fbe-4d78-af88-8f4a29266b77.png">

⏫ Также у тебя не совсем удачно экспортирована иконка из макета. Обращай внимание на уровень вложенности в макете при выборе иконки, чтоб достать ресурс правильно.
Ты видимо достал так
<img width="305" alt="image" src="https://user-images.githubusercontent.com/22057568/219954707-054c26ca-c502-43ff-ab5a-76e5486eabba.png">

А надо так, чтоб достать целиком иконку, как задумывал автор макета. Иконки обычно всегда квадратные
<img width="314" alt="image" src="https://user-images.githubusercontent.com/22057568/219954743-5cae4a98-168c-4b35-91d6-432d173a6915.png">

Если корректно заэкспортировать все иконки (рекомендую пересмотреть все в твоем проекте), то будет легче решить проблемы с отступами

SeveNChaK commented on app/src/main/res/layout/track_card.xml:
```diff
@@ -1,38 +1,60 @@
+            android:textFontWeight="400"
+            android:textSize="@dimen/settings_text_size" />
 
         <LinearLayout
```
⚠️ Все еще остается проблема с длинным название артиста (время трека задал 14:33, но вторая 3 обрезалась)
<img width="407" alt="image" src="https://user-images.githubusercontent.com/22057568/219954941-c980836c-099f-4bf7-855a-3722dc20de5c.png">

Чтобы это исправить, можно убрать или задать = 0 аттрибут `android:layout_weight` у вью времени и выставить `android:layout_width` в `wrap_content` у этой же вью. Тогда ничего не будет обрезаться с длинным название
<img width="407" alt="image" src="https://user-images.githubusercontent.com/22057568/219955133-cb60531a-d686-41c2-95bf-0cee0da50f9b.png">
и с коротким тоже будет хорошо работать
<img width="402" alt="image" src="https://user-images.githubusercontent.com/22057568/219955151-35160c57-f594-458e-a3df-f30b13d4334a.png">

SeveNChaK commented on app/src/main/res/layout/track_card.xml:
```diff
@@ -1,38 +1,60 @@
-<LinearLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
-    style="@style/main_track_card_style">
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
⚠️ У списка лишний отступ справа и слева
<img width="420" alt="image" src="https://user-images.githubusercontent.com/22057568/219955543-938333d9-bb78-44eb-a4c6-e06600d63674.png">
В макете у элементо по бокам такой же отступ как и у поля ввода
<img width="607" alt="image" src="https://user-images.githubusercontent.com/22057568/219955652-d3c82937-2a60-4aba-9333-f4165b732ccc.png">

⚠️ Не хваатет отступа от поле ввода и первого элемента списка
Сейчас он идет практически сразу
<img width="419" alt="image" src="https://user-images.githubusercontent.com/22057568/219955611-301db6c1-c6ce-42b8-98e4-44ce5b7d9c11.png">

В макете
<img width="642" alt="image" src="https://user-images.githubusercontent.com/22057568/219955588-13933766-b26f-48bb-9a68-7daa475d1d38.png">

SeveNChaK reviewed (commented):
⏫ Также осталось с прошлых спринтов или отвалилось по ходу дела, но отступ между тулбаром и полем ввода текст на макете меньше. Это про экран поиска
Сейчас в приложении
<img width="445" alt="image" src="https://user-images.githubusercontent.com/22057568/219955460-82ead338-011e-4e41-a67e-6be8ef082b77.png">
Макет
<img width="626" alt="image" src="https://user-images.githubusercontent.com/22057568/219955446-c7a6eaaa-71d2-428a-a1e0-2a7d2e37d712.png">

PavelRomashin committed:
отработка замечаний к дз10:
1)поправлены отступы в соответствии с замечаниями
2) fill_parent заменены на wrap_content
3) поправлено отображение информации о треке в части корректного отображения длительности песни
4) перезагрузил иконки стрелочки вперед и поделиться

SeveNChaK commented on app/src/main/res/layout-v28/activity_search.xml:
```diff
@@ -18,8 +18,8 @@
         android:orientation="vertical"
-        android:paddingHorizontal="16dp"
-        android:paddingVertical="36dp"
+        android:paddingHorizontal="8dp"
```
не правильный отступ, долежн быть 16
